options/path.c

```diff
@@ -315,12 +316,15 @@ bool mp_is_url(bstr path)
     int proto = bstr_find0(path, "://");
     if (proto < 1)
         return false;
-    // The protocol part must be alphanumeric, otherwise it's not an URL.
+    // Per RFC3986, the first character of the protocol must be alphanumeric, 
+    // the rest must be alphanumeric plus - + and .
     for (int i = 0; i < proto; i++) {
         unsigned char c = path.start[i];
-        if (!(c >= 'a' && c <= 'z') && !(c >= 'A' && c <= 'Z') &&
-            !(c >= '0' && c <= '9') && c != '_')
```

Comment:
does the removal of `_` here break any existing usecases?

Reply:
the ` c != '_'` is still there (line 325), though i wonder why if it isn't int he spec?

Reply:
I can't think of anything that could be using it, but I erred on the side of caution since I don't think it being there could break anything.

Reply:
So is the `c != '_'` supposed to be there or not? because it isn't.

Reply:
I think we should merge this without "_" and only add it if we find a valid use-case that requires non-spec behavior.